### PR TITLE
SW-929: fix translation status logic and add test coverage

### PR DESCRIPTION
**Pull request by wheelsandcogs**
We keep getting edge cases where the translation status trips users up - this is an attempt to simplify the logic and adds test cases so we hopefully don't regress in future.

### CODE REVIEW

File: src/routes/developer.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ devRouter.use((req, res, next) => {
     next(new ForbiddenException('user is not a developer'));
     return;
   }
-  logger.info(`user is a developer`);
+  logger.info(`user ${req.user?.id} is a developer`);
```
Comment (wheelsandcogs) [inline src/routes/developer.ts:16]: unrelated, but might as well stay in now

File: src/dtos/tasklist-state-dto.ts [hunk 2]
```diff
@@ -180,88 +179,66 @@ export class TasklistStateDTO {
...
-      return every(translatableMetadataKeys, (key) => {
-        // ignore roundingDescription if rounding isn't applied, otherwise check some data exists
-        return key === 'roundingDescription' && !revision.roundingApplied ? true : Boolean(meta[key]);
-      });
-    });
-
-    const relatedLinksTranslated = every(revision.relatedLinks, (link) => {
-      return link.labelEN && link.labelCY;
-    });
+    const changesSinceImport = currentTranslations.some((t: TranslationDTO) => {
```
[the remark this answers is not in the document]
Comment (wheelsandcogs) [reply src/dtos/tasklist-state-dto.ts:208]: Doesn't matter, the validation on the import will flag unknown keys.
